components/style/properties/declaration_block.rs

```diff
@@ -751,6 +755,7 @@ impl ToCss for PropertyDeclarationBlock {
                              importance,
                              &mut is_first_serialization)?;
                     } else {
+                        shorthands_used.push(shorthand);
```

Comment:
I believe it would be enough to add all the longhands in `shorthand.longhands()` to `already_serialized`, wouldn't it?